dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -1584,15 +1584,20 @@ void
 Sedp::Writer::end_historic_samples(const DCPS::RepoId& reader)
 {
   const void* pReader = static_cast<const void*>(&reader);
-  ACE_Message_Block mb(DCPS::DataSampleHeader::max_marshaled_size(),
-                       ACE_Message_Block::MB_DATA,
-                       new ACE_Message_Block(static_cast<const char*>(pReader),
-                                             sizeof(reader)));
-  mb.set_flags(ACE_Message_Block::DONT_DELETE);
```

Comment:
Since DONT_DELETE is no longer used, update the methods control_delivered() and control_dropped() to simplify them.